File: src/webui/app/logic/domain/time-plan-activity-feasability.ts
import { TimePlanActivityFeasability } from "@jupiter/webapi-client";

export function timePlanActivityFeasabilityName(
  feasability: TimePlanActivityFeasability
): string {
  switch (feasability) {
    case TimePlanActivityFeasability.MUST_DO:
      return "Must Do";
    case TimePlanActivityFeasability.NICE_TO_HAVE:
      return "Nice To Have";
    case TimePlanActivityFeasability.STRETCH:
      return "Stretch";
  }
}

const TIME_PLAN_ACTIVITY_FEASABILITY_MAP = {
  [TimePlanActivityFeasability.MUST_DO]: 0,
  [TimePlanActivityFeasability.NICE_TO_HAVE]: 1,
  [TimePlanActivityFeasability.STRETCH]: 2,
};

export function compareTimePlanActivityFeasability(
  feasability1: TimePlanActivityFeasability,
  feasability2: TimePlanActivityFeasability
): number {
  return (
    TIME_PLAN_ACTIVITY_FEASABILITY_MAP[feasability1] -
    TIME_PLAN_ACTIVITY_FEASABILITY_MAP[feasability2]
  );
}

File: src/core/jupiter/core/domain/concept/time_plans/time_plan_activity_feasability.py
"""The feasability of a particular activity within a plan."""

from functools import total_ordering

from jupiter.core.framework.value import EnumValue, enum_value


@total_ordering
@enum_value
class TimePlanActivityFeasability(EnumValue):
    """The feasability of a particular activity within a plan."""

    MUST_DO = "must-do"
    NICE_TO_HAVE = "nice-to-have"
    STRETCH = "stretch"

    def __lt__(self, other: object) -> bool:
        """Compare this with another."""
        if not isinstance(other, TimePlanActivityFeasability):
            raise Exception(
                f"Cannot compare with {other.__class__.__name__}",
            )

        all_values = self.get_all_values()

        return all_values.index(self.value) < all_values.index(other.value)
